test/tstLinearBVH.cpp

```diff
@@ -350,6 +350,158 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(not_exceeding_stack_capacity, DeviceType,
   BOOST_TEST(ArborX::lastElement(offset) == n);
 }
 
+template <typename DeviceType>
+struct CustomCallbackSpatialPredicate
+{
+  using tag = ArborX::Details::InlineCallbackTag;
+  Kokkos::View<ArborX::Point *, DeviceType> points;
+  ArborX::Point const origin = {{0., 0., 0.}};
+  template <typename Query, typename Insert>
+  KOKKOS_FUNCTION void operator()(Query const &, int index,
+                                  Insert const &insert) const
+  {
+    float const distance_to_origin =
+        ArborX::Details::distance(points(index), origin);
+    insert({index, distance_to_origin});
+  }
+};
+
+template <typename DeviceType>
+struct CustomCallbackSpatialPredicate2
+{
+  using tag = ArborX::Details::PostCallbackTag;
+  Kokkos::View<ArborX::Point *, DeviceType> points;
+  ArborX::Point const origin = {{0., 0., 0.}};
+  template <typename InOutView, typename InView, typename OutView>
+  void operator()(InOutView &offset, InView in, OutView &out) const
+  {
+    using ExecutionSpace = typename DeviceType::execution_space;
+    using ArborX::Details::distance;
+    auto const n = offset.extent(0) - 1;
+    ArborX::reallocWithoutInitializing(out, in.extent(0));
+    auto points_ = points;
+    auto origin_ = origin;
```

Comment:
Why do you do this?

Reply:
To avoid implicit capture of `*this`.  I will add a comment.